cli/kube_deployment/kube_deployment.go

```diff
@@ -53,11 +54,26 @@ func (p *KubeDeploymentConfigPlugin) Sign(dep map[string]interface{}, privKey *r
 	}
 
 	// Get the base 64 encoding of the kube operator, and put it into the deployment config.
-	if b64, err := ConvertFileToB64String(operatorFilePath); err != nil {
+	b64, err := ConvertFileToB64String(operatorFilePath)
+	if err != nil {
 		return true, "", "", errors.New(msgPrinter.Sprintf("unable to read kube operator %v, error %v", dep["operatorYamlArchive"], err))
-	} else {
-		dep["operatorYamlArchive"] = b64
 	}
+	dep["operatorYamlArchive"] = b64
+
+	if _, ok := dep["metadata"]; ok {
```

Comment:
How to distinguish the cases when the service is first created or updated. For the update case, "metadata" might exist already.

Reply:
@linggao So `SignDeployment` function is called [here](https://github.com/open-horizon/anax/blob/5b89c0d877975b965f9451d099d54bef0bd9382f/cli/exchange/service.go#L195), which calls `Sign` function. It only checks the metadata field in service definition file. 
Then it will check if service is already exist. Update service if exist, or create service if not exist.
Should be fine for update service case, right?